OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs

```diff
@@ -126,6 +115,17 @@ void ITick.Tick(Actor self)
 			if (self.Disposed)
 				return;
 
+			// Defer updating the frozen actor until we are sure that the
+			// actor's ITargetablePositions traits have been initialized
+			if (!created && startsRevealed)
```

Comment:
We can probably merge these two bools into a single `revealOnFirstTick`

Reply:
Are you sure about that? `created` gets checked in `OnVisibilityChanged` by itself. Not sure if it would still do the right thing if those were merged.

Reply:
I wasn't sure, so I guess not.